internal/mode/static/nginx/agent/command.go

```diff
@@ -40,50 +59,99 @@ func (cs *commandService) CreateConnection(
 	}, nil
 }
 
+// Subscribe is a decoupled communication mechanism between the data plane agent and control plane.
 func (cs *commandService) Subscribe(in pb.CommandService_SubscribeServer) error {
-	fmt.Println("Received subscribe request")
-
 	ctx := in.Context()
 
+	gi, ok := grpcContext.GrpcInfoFromContext(ctx)
+	if !ok {
+		return agentgrpc.ErrStatusInvalidConnection
+	}
+
+	cs.logger.Info(fmt.Sprintf("Received subscribe request from %q", gi.IPAddress))
+
+	go cs.listenForDataPlaneResponse(ctx, in)
+
+	// wait for connection to be established
+	podName, err := cs.waitForConnection(ctx, gi)
```

Comment:
Do we want to couple CreateConnection and Subscribe like this? Are they dependent on one another? Will the agent try to Subscribe again if we timeout waiting for the connection?

Reply:
The names are a bit confusing here. CreateConnection really is just a way for agent to send information about itself to us. Subscribe doesn't require it to be called. However, we need it in order to link a subscription with the proper agent, because the Subscribe call doesn't really tell us which agent is connecting (other than IP address).

Reply:
I see. So what does the agent do if we timeout here?

Reply:
The agent just sits and waits for a message from us. If it receives this error, I believe it prints it out and then sends out a new Subscribe call.

Reply:
That's good. I'm surprised there's no metadata that we can use to link the Subscribe to an agent, like a uuid or something like that. We are creating a dependency here between the two calls that I don't think the agent team anticipated.

Reply:
I'm not a fan of this structure either. When we support auth, we can inject the uuid into the context for access in the Subscribe call. The problem still exists though where we don't know which Pod that uuid belongs to until we can link it to the hostname received in a CreateConnectionRequest.

I'm guessing the intention was that they are independent because a control plane can ask for a health status update from the agent right away through a subscribe channel. Meanwhile, an agent takes a bit to discover its local nginx instance, and then sends a specific call to the control plane to register it.